File: 202504/utils/ruby/validity_check.rb
#!/usr/bin/env ruby
# performance_measurement.rb
# Grep、Sort、HashMapの各アルゴリズムのパフォーマンスと正当性を計測するプログラム

# 正当性検証ユーティリティ
class ValidityCheck
  # 結果が期待値と一致するかチェック
  def self.verify(name, result, expected)
    if result == expected
      puts "#{name} 正当性検証: 成功 ✓"
      return true
    else
      puts "#{name} 正当性検証: 失敗 ✗"
      puts "  期待値: #{expected.inspect}"
      puts "  実際の結果: #{result.inspect}"
      return false
    end
  end
end

File: 202504/utils/javascript/validity_check.js
/**
 * 結果が期待値と一致するか検証する
 * @param {string} name - テスト名
 * @param {any} result - 実際の結果
 * @param {any} expected - 期待値
 * @returns {boolean} 検証結果
 */
export function verifyResult(name, result, expected) {
  // 深い比較
  const isEqual = JSON.stringify(result) === JSON.stringify(expected);
  
  if (isEqual) {
    console.log(`${name} 正当性検証: 成功 ✓`);
    return true;
  } else {
    console.log(`${name} 正当性検証: 失敗 ✗`);
    console.log(`  期待値: `, expected);
    console.log(`  実際の結果: `, result);
    return false;
  }
}
